library/ssl_tls.c

```diff
@@ -1704,6 +1704,17 @@ void mbedtls_ssl_tls13_conf_early_data( mbedtls_ssl_config *conf,
 {
     conf->early_data_enabled = early_data_enabled;
 }
+
+#if defined(MBEDTLS_SSL_SRV_C)
+void mbedtls_ssl_tls13_conf_max_early_data_size(
+         mbedtls_ssl_config *conf, uint32_t max_early_data_size )
+{
+    conf->max_early_data_size =
+        max_early_data_size < MBEDTLS_SSL_MAX_EARLY_DATA_SIZE ?
+                        max_early_data_size : MBEDTLS_SSL_MAX_EARLY_DATA_SIZE;
```

Comment:
Just
```suggestion
    conf->max_early_data_size = max_early_data_size;
```
now I think. Otherwise `conf->max_early_data_size` needs to be set to its default value MBEDTLS_SSL_MAX_EARLY_DATA_SIZE somewhere.

Reply:
> now I think. Otherwise `conf->max_early_data_size` needs to be set to its default value MBEDTLS_SSL_MAX_EARLY_DATA_SIZE somewhere.

`ssl_tls13_update_early_data_state()` from https://github.com/Mbed-TLS/mbedtls/pull/6719/commits/964086d551fd14f7fe640ea1b8166168cfb73530 might be a good place. I will add it in #6719.

Reply:
What about `mbedtls_ssl_config_defaults()` ? This seems the right (dedicated ?) function.

Reply:
https://github.com/Mbed-TLS/mbedtls/pull/6719/commits/4be7f5eb3499ac4ce9365d2e34883b501afba95f , just add it in this commit

Reply:
But MBEDTLS_SSL_MAX_EARLY_DATA_SIZE is not a limit anymore, just the default value, right ? Thus `mbedtls_ssl_tls13_conf_max_early_data_size()` can be called with a value greater than MBEDTLS_SSL_MAX_EARLY_DATA_SIZE. That's valid and we should not overwrite it. Setting `conf->max_early_data_size` to MBEDTLS_SSL_MAX_EARLY_DATA_SIZE in `mbedtls_ssl_config_defaults()` seems the right choice to me here.

Reply:
> What about `mbedtls_ssl_config_defaults()` ? This seems the right (dedicated ?) function.

It is not right place. That's the place for setting default value,  not right place for checking limitation. And `mbedtls_ssl_conf_max_early_data_size()` is called after `mbedtls_ssl_config_defaults()` like others `mbedtls_ssl_conf_*`. That can not guarantee the value.  And in my experience, `mbedtls_ssl_conf_*` can be called after `mbedtls_ssl_setup()` also.

I think we should add default value in `mbedtls_ssl_config_defaults()` for `early_data_enabled` and `max_early_data_size` . I will put it in this PR.

Reply:
> I think we should add default value in `mbedtls_ssl_config_defaults()` for `early_data_enabled` and `max_early_data_size` . I will put it in this PR.

Done in https://github.com/Mbed-TLS/mbedtls/pull/6537/commits/18415e830a502f8e8eb1f78613a7618a10ec1036